health.go

```diff
@@ -97,17 +98,24 @@ func (h *health) scheduleCheck(task *checkTask, cfg *Config) {
 		if !h.runCheckOrStop(task, time.After(cfg.InitialDelay)) {
 			return
 		}
-
+		h.reportResults()
 		// scheduled recurring execution
 		task.ticker = time.NewTicker(cfg.ExecutionPeriod)
 		for {
 			if !h.runCheckOrStop(task, task.ticker.C) {
 				return
 			}
+			h.reportResults()
 		}
 	}()
 }
 
+func (h *health) reportResults() {
+	h.lock.RLock()
+	defer h.lock.RUnlock()
+	h.healthListener.OnResultsUpdated(copyResultsMap(h.results))
```

Comment:
In the odd case that this takes slightly longer than expected, I'd copy the results first, release the lock and then call the listener.